packages/extension/src/utils/storageChromeLocal.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Manages the storage in Chrome local storage
+ */
+
+export const saveInChromeLocalStorage = (key: string, value: any): Promise<void> => {
```

Comment:
Why do you need it to return a promise?

What about something like this? You can use the helpers `isString`, `isNumber` etc. that I wrote in my previous PR.

```jsx
/**
 * Set an item (key/value) in local storage.
 *
 * @param {string} key The key
 * @param {unknown} value The value
 */
export const setLocalStorageItem = (
  key: string,
  value: Record<string, any> | string | number | boolean,
) => {
  try {
    if (isString(value)) {
      localStorage.setItem(key, value)
    } else if (isNumber(value) || isBoolean(value)) {
      localStorage.setItem(key, value.toString())
    } else if (isObject(value)) {
      localStorage.setItem(key, JSON.stringify(value))
    } else {
      console.error(`Unrecognized local storage value ${value} for key: ${key}`)
    }
  } catch (err) {
    console.error(err)
  }
}
```

Reply:
Your code sample is for localStorage whici is synchronous but here we use Chrome.storage.local which is async by nature. I can see if the code can be changed, but keep in mind it is not the same storage